lib/ansiblelint/__main__.py

```diff
@@ -84,24 +84,30 @@ def choose_formatter_factory(
     return r
 
 
-def report_outcome(matches: List["MatchError"], options) -> int:
+def report_outcome(matches: List["MatchError"], checked_files: Set[str], options) -> int:
     """Display information about how to skip found rules.
 
     Returns exit code, 2 if errors were found, 0 when only warnings were found.
     """
-    failure = False
+    failures = 0
+    warnings = 0
     msg = """\
 You can skip specific rules or tags by adding them to your configuration file:
 ```yaml
 # .ansible-lint
 warn_list:  # or 'skip_list' to silence them completely
 """
-
+    # counting
     matched_rules = {match.rule.id: match.rule for match in matches}
+    for match in matches:
+        if {match.rule.id, *match.rule.tags}.isdisjoint(options.warn_list):
+            failures += 1
+        else:
+            warnings += 1
+
     for id in sorted(matched_rules.keys()):
         if {id, *matched_rules[id].tags}.isdisjoint(options.warn_list):
             msg += f"  - '{id}'  # {matched_rules[id].shortdesc}\n"
-            failure = True
     for match in matches:
```

Comment:
do we really need to iterate over matches again?

Reply:
That is not the right place to look for performance improvements. It is written as is for code readability, which is more important in this case.  Even in the worst case possible of having >1000 violations found, the cost is likely measures in few milliseconds.

Looking for performance is good, there are places where this really matters (see recent fix around parsing files multiple times, which made a 10x speed improvement).